### PR TITLE
Develop upstream sync 180827

### PR DESCRIPTION
**Pull request by sunway513**
Don't Merge! 
Weekly sync with upstream, this PR is to kick off PR.

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/elemental_ir_emitter.h [hunk 1]
```diff
@@ -100,7 +100,7 @@ class GpuElementalIrEmitter : public ElementalIrEmitter {
       tensorflow::gtl::ArraySlice<llvm::Value*> operands,
       tensorflow::gtl::ArraySlice<PrimitiveType> input_type,
       PrimitiveType output_type,
-      tensorflow::gtl::ArraySlice<llvm::Attribute::AttrKind> attributes);
+      tensorflow::gtl::ArraySlice<llvm::Attribute::AttrKind>     attributes,  llvm::IRBuilder<>* ir_builder, llvm::Module* modu    le);
```
Comment (whchung) [inline tensorflow/compiler/xla/service/gpu/elemental_ir_emitter.h]: typo?

**Comment by sunway513** [comment]
jenkins: retest this please

**Comment by sunway513** [comment]
The CI failed with the following error messages: 
```
tensorflow/core/kernels/relu_op_gpu.cu.cc:47:5: error: unknown type name 'half2'
    half2 gradient_h2 = reinterpret_cast<const half2*>(gradient)[index];
    ^
```
CC @deven-amd

**Comment by deven-amd** [comment]
@sunway513 @whchung, 

this is weird....the line that errors out has not changed in months, neither have we changes anything regarding hip fp16 support recently.  Looking into it, but coming up empty so far

**Comment by whchung** [comment]
@sunway513 

- on ROCm path `xla::gpu::GpuElementalIrEmitter::EmitDeviceFunctionCall` is missing.
- on CUDA path:
  - `tensorflow/compiler/xla/ptr_util.h` seems to be missing somehow
  - `tensorflow/core/kernels/relu_op_gpu.cu.cc` seems to have error. seems terminology change wasn't complete

**Comment by whchung** [comment]
it seems errors on both paths still exist

**Comment by whchung** [comment]
@sunway513 On CUDA path two failed cases:

```
//tensorflow/python/kernel_tests:softplus_op_test
//tensorflow/python/kernel_tests:softsign_op_test
```

Please help review full build/test log to see if whitelist shall be modified.

ROCm path is still undergoing.

**Pull request merged**